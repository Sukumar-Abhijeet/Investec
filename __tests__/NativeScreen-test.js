import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import NativeScreen from '../src/@Screens/Auth/Native';

const mockStore = configureMockStore();
const store = mockStore({userData:{userName:"Sukumar Abhijeet"}});

const createTestProps = {
  navigation: {
    state: { params: {} },
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn()
  }
};

test('Testing Button Screen .js', () => {
  const snap = renderer.create(
      <Provider store={store}>
        <NativeScreen  {...createTestProps} />
      </Provider>
  ).toJSON();
  expect(snap).toMatchSnapshot();
});