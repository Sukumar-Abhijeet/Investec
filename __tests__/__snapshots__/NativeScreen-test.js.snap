// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button Screen .js 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 15.714285714285714,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "flex-end",
        "color": "#000",
        "fontSize": 22,
        "fontWeight": "bold",
        "margin": 31.428571428571427,
      }
    }
  >
    Welcome - 
    Sukumar Abhijeet
  </Text>
  <View
    style={
      Object {
        "alignSelf": "center",
        "bottom": 78.57142857142857,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#6eb1f7",
          "borderRadius": 15.714285714285714,
          "marginBottom": 15.714285714285714,
          "opacity": 1,
          "paddingVertical": 23.57142857142857,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Visit Buttons
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#6eb1f7",
          "borderRadius": 15.714285714285714,
          "marginBottom": 15.714285714285714,
          "opacity": 1,
          "paddingVertical": 23.57142857142857,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 22,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Check Login
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
